src/librbd/internal.cc

```diff
@@ -1831,14 +1890,28 @@ namespace librbd {
 	Context *req_comp = new C_ContextCompletion(*completion);
 	librados::AioCompletion *rados_completion =
 	  librados::Rados::aio_create_completion(req_comp, NULL, rados_ctx_cb);
+
+	bool flag_nonexistent = false;
 	if (p->offset == 0) {
+	  // TODO: JD if the image as snapshots, do we need to keep the
+	  // object flagged as existing? How does RADOS know to delete
+	  // old snapshot objects?
+	  flag_nonexistent = true;
+	  m_ictx->update_object_map(p->objectno, p->objectno + 1,
```

Comment:
@jdurgin The current shorter overload doesn't include the current state filter.  I can add the current state filter and default it to empty.